S7.Net/PlcSynchronous.cs

```diff
@@ -347,43 +321,18 @@ public void WriteClass(object classValue, int db, int startByteAdr = 0)
             WriteClassAsync(classValue, db, startByteAdr).GetAwaiter().GetResult();
         }
 
-        private void ReadBytesWithSingleRequest(DataType dataType, int db, int startByteAdr, byte[] buffer, int offset, int count)
-        {
-            try
-            {
-                // first create the header
-                int packageSize = 19 + 12; // 19 header + 12 for 1 request
-                var package = new System.IO.MemoryStream(packageSize);
-                BuildHeaderPackage(package);
-                // package.Add(0x02);  // datenart
-                BuildReadDataRequestPackage(package, dataType, db, startByteAdr, count);
-
-                var dataToSend = package.ToArray();
-                var s7data = RequestTsdu(dataToSend);
-                AssertReadResponse(s7data, count);
-
-                Array.Copy(s7data, 18, buffer, offset, count);
-            }
-            catch (Exception exc)
-            {
-                throw new PlcException(ErrorCode.ReadData, exc);
-            }
-        }
-
-#if NET5_0_OR_GREATER
-
         private void ReadBytesWithSingleRequest(DataType dataType, int db, int startByteAdr, Span<byte> buffer)
         {
             try
             {
                 // first create the header
-                int packageSize = 19 + 12; // 19 header + 12 for 1 request
-                var package = new System.IO.MemoryStream(packageSize);
+                const int packageSize = 19 + 12; // 19 header + 12 for 1 request
+                var dataToSend = new byte[packageSize];
+                var package = new MemoryStream(dataToSend);
```

Comment:
Instead of a `ToArray` on the MS we can just give it our buffer and use that buffer later on.
Avoids allocations (of the MS buffer).